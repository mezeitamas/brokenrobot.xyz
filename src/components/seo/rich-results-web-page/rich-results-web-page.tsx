import React from 'react';
import type { FunctionComponent } from 'react';

import type { Person, Organization, WithContext, WebPage } from 'schema-dts';

type RichResultsWebPageProps = {
    siteName: string;
    siteUrl: string;
    author: {
        name: string;
        social: {
            github: string;
            linkedin: string;
        };
    };
};

const RichResultsWebPage: FunctionComponent<RichResultsWebPageProps> = ({ siteName, siteUrl, author }) => {
    const authorPerson: Person = {
        '@type': 'Person',
        '@id': `${siteUrl}/about-me/#Person`,
        name: author.name,
        url: `${siteUrl}/about-me/`,
        sameAs: [author.social.github, author.social.linkedin]
    };

    const publisher: Organization = {
        '@type': 'Organization',
        '@id': `${siteUrl}/#Organization`,
        name: siteName,
        url: siteUrl
    };

    const webPage: WithContext<WebPage> = {
        '@context': 'https://schema.org',
        '@type': 'WebPage',
        '@id': `${siteUrl}/#WebPage`,
        name: siteName,
        url: siteUrl,
        inLanguage: 'en-US',
        author: [authorPerson],
        publisher: publisher
    };

    return <script type="application/ld+json">{JSON.stringify(webPage)}</script>;
};

export { RichResultsWebPage };

export type { RichResultsWebPageProps };
