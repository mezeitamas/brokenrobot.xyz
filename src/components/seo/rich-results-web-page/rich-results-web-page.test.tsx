import React from 'react';

import { create } from 'react-test-renderer';

import { RichResultsWebPage } from './rich-results-web-page';
import type { RichResultsWebPageProps } from './rich-results-web-page';

describe('<RichResultsWebPage />', () => {
    test('should render', () => {
        // Given
        const props: RichResultsWebPageProps = {
            siteName: 'Broken Robot',
            siteUrl: 'https://www.brokenrobot.xyz',
            author: {
                name: 'Tamas Mezei',
                social: {
                    githubUrl: 'https://github.com/mezeitamas',
                    linkedinUrl: 'https://www.linkedin.com/in/mezeitamas/',
                    twitterUrl: 'https://twitter.com/brokenrobot_xyz',
                    twitterUsername: '@brokenrobot_xyz'
                }
            }
        };

        // When
        const component = create(<RichResultsWebPage {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
