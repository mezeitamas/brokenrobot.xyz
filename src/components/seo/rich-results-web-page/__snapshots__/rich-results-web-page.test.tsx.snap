// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichResultsWebPage /> should render 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"WebPage","@id":"https://www.brokenrobot.xyz/#WebPage","name":"Broken Robot","url":"https://www.brokenrobot.xyz","inLanguage":"en-US","author":[{"@type":"Person","@id":"https://www.brokenrobot.xyz/about-me/#Person","name":"Tamas Mezei","url":"https://www.brokenrobot.xyz/about-me/","sameAs":["https://github.com/mezeitamas","https://www.linkedin.com/in/mezeitamas/"]}],"publisher":{"@type":"Organization","@id":"https://www.brokenrobot.xyz/#Organization","name":"Broken Robot","url":"https://www.brokenrobot.xyz"}}
</script>
`;
